### PR TITLE
Svg color schema fixes

### PR DESCRIPTION
**Pull request by daschuer**
This fixes some remaining issues with SVGs, color schemes and scaling.
Now the DarkMetal skin looks nice again. 
Fixes https://bugs.launchpad.net/mixxx/+bug/1686663

This includes also some refactoring to remove old unused code.

### CODE REVIEW

File: src/widget/paintable.cpp [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+    if (!source.isSVG()) {
+        m_pPixmap.reset(WPixmapStore::getPixmapNoCache(source.getPath(), scaleFactor));
+    } else {
+        m_pSvg.reset(new QSvgRenderer());
+        if (source.getData().isEmpty()) {
+            m_pSvg->load(source.getPath());
+        } else {
+            m_pSvg->load(source.getData());
+        }
+        if (mode == TILE || mode == Paintable::FIXED || WPixmapStore::willCorrectColors()) {
```
Comment (rryan) [inline src/widget/paintable.cpp]: This change converts all FIXED SVGs to raster? Won't this cause blurry SVGs on hidpi displays?

**Comment by daschuer** [comment]
No, because it creates a 1:1 target pixmap. It was done this way before, this PR move the code a bit around to make it more obvious. For qt5 this will indeed an issue if we not   consider the device pixels ... a topic for Mixxx 2.2.

**Comment by rryan** [comment]
> No, because it creates a 1:1 target pixmap. It was done this way before, this PR move the code a bit around to make it more obvious. For qt5 this will indeed an issue if we not consider the device pixels ... a topic for Mixxx 2.2.

Do you mean it's ok because scaleFactor scales the target pixmap to 2x? That's not the case if you are either using Qt5 or are on macOS on Qt4.

**Comment by daschuer** [comment]
> Do you mean it's ok because scaleFactor scales the target pixmap to 2x? That's not the case if you are either using Qt5 or are on macOS on Qt4.

Yes. But I was not aware, that macOs already does the scaling. 

I now remember that we have enabled it for 2.0 at a time. How plays the new scale factor on MacOs? 
Should we revert the scaling auto setting because we now have our own? 

Unfortunately I cannot test it. Would you mind to adopt this issue and propose a solution for MacOS?

File: src/widget/paintable.cpp (hunk 1 above)
Comment (daschuer) [inline src/widget/paintable.cpp]: Fixed in the latest commit.

**Comment by daschuer** [comment]
@Be-ing the concerns from RJ are issued. Can you pleas have a loop if it works for you?
Thank you.

**Comment by Be-ing** [comment]
I tested with #1179 and it looks good to me. I can't test whether scaling looks good on Mac though.

**Comment by daschuer** [comment]
Ok then I will merge it and hope (or actually hope not) we master receive test results form Mac user. 
Thank you for looking into it.

**Pull request merged**